crates/driver/src/domain/competition/mod.rs

```diff
@@ -413,6 +512,13 @@ fn merge(solutions: impl Iterator<Item = Solution>, auction: &Auction) -> Vec<So
     merged
 }
 
+struct SettleRequest {
+    auction_id: Option<i64>,
```

Comment:
Can we avoid having this optional? Looks like an oversight when the auction id was introduced into the `/settle` request. The autopilot now always sends it so there is no need to have it optional in the first place.

Reply:
Isn't that required for a 2-step migration? https://github.com/cowprotocol/services/pull/3131

Reply:
Ahh, we are back to the issue that un-updated drivers we don't run still have the `deny_unknown_fields` on the DTOs.
Yeah, in that case this needs to remain optional. :/